src/seedu/addressbook/Main.java

```diff
@@ -87,6 +95,17 @@ private void runCommandLoopUntilExitCommand() {
             ui.showResultToUser(result);
 
         } while (!ExitCommand.isExit(command));
+
+        printTagSessionHistory();
+    }
+
+    /** Prints tagging history done during the session */
+    private void printTagSessionHistory() {
+        ui.showToUser(TAGSESSIONHISTORY_HEADER);
+        for (Tagging tagging : addressBook.getTagSessionHistory()) {
+            String action = tagging.getAction()==TagAction.ADD? TAGSESSIONHISTORY_ADD_ACTION_SYMBOL : TAGSESSIONHISTORY_REMOVE_ACTION_SYMBOL;
```

Comment:
Remember to have a space before and after your operators, in this case `==` and `?`.